src/Utilities/copy.jl

```diff
@@ -359,15 +358,30 @@ function _cost_of_bridging(
 ) where {S<:MOI.AbstractVectorSet}
     x = MOI.get(dest, MOI.VariableBridgingCost{S}())
     y = MOI.get(dest, MOI.ConstraintBridgingCost{MOI.VectorOfVariables,S}())
-    # In case of ties, we give priority to vector sets. See issue #987
-    return (x - y, true, x + y)
+    return !iszero(x), x - y, false
 end
 
 """
     sorted_variable_sets_by_cost(dest::MOI.ModelLike, src::MOI.ModelLike)
 
-Returns a `Vector{Type}` of the set types corresponding to `VariableIndex` and
+Return a `Vector{Type}` of the set types corresponding to `VariableIndex` and
 `VectorOfVariables` constraints in the order in which they should be added.
+
+The order is important because some solvers require variables to be added in
+particular order, and the order can also impact the bridging decisions.
+
+The sorting happens in the `_cost_of_bridging` function and has three main
+considerations:
+
+1. First add sets for which the `VariableBridgingCost` is `0`. This ensures that
+   we minimize the number of variable bridges that get added. (Variable bridges
+   are bad because they produce an expression that needs substituting into the
```

Comment:
> Variable bridges are bad because they produce an expression that needs substituting into the remainder of the model

What does "bad" mean ? It's a bit subjective and it's unclear why substitution is bad (it's not adding any additional variable or constraint so the reader why thing instead that it's quite good).
I would say that we give priority to passing the model as-is, without transformation. Even if the `ConstraintBridgingCost` is zero as well, the constraint will need to be forced-bridged into affine-in-set

Reply:
I removed the sentence.